src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -186,11 +202,76 @@ public DynamoView(DynamoViewModel dynamoViewModel)
                  }
              };
 
+            // Add an event when the collection is modified.   
+            tabItems.CollectionChanged += this.OnCollectionChanged;
+
+            this.HideOrShowRightSideBar();
+
             this.dynamoViewModel.RequestPaste += OnRequestPaste;
             this.dynamoViewModel.RequestReturnFocusToView += OnRequestReturnFocusToView;
             FocusableGrid.InputBindings.Clear();
         }
 
+        // This method adds a tab item to the right side bar and 
+        // sets the extension window as the tab content.
+        internal TabItem AddTabItem(ContentControl contentControl)
+        {
+            int count = TabItems.Count;
+
+            tabDynamic.DataContext = null;
+
+            // creates a new tab item
+            TabItem tab = new TabItem();
+            tab.Header = ((Window)contentControl).Title;
```

Comment:
hmm - this is not great - you cannot expect that the contentControl is a window, what about if we think about this a bit more. 

a few questions for you to consider

* Should one extension be able to add more than on tab? - right now they can.
* Should an extension be able to decide its tab title or should it be implied from the extension name?
* Should an extension author call/pass the content directly to this method or should they implement a method which returns content and we do the rest?

Reply:
I agree, just to show some title I made this line. I was thinking, setting the title of the tab with the value the user passes to the extension name would be a good start. In that case, this line would be removed and will be using the name passed by the user.